src/plays/self-clicking-button/SelfClickingButton.js

```diff
@@ -0,0 +1,49 @@
+import PlayHeader from 'common/playlists/PlayHeader';
+import './styles.css';
+import { useState, useEffect, useRef } from 'react';
+import { format } from 'date-fns';
+
+// WARNING: Do not change the entry componenet name
+function SelfClickingButton(props) {
+  // Your Code Start below.
+  const buttonRef = useRef(null);
+  const [counter, setCounter] = useState(0);
+
+  function clickHandler(event) {
+    setCounter((prev) => prev + 1);
+  }
+
+  useEffect(() => {
+    buttonRef.current.addEventListener('click', clickHandler);
+
+    const interval = setInterval(() => {
+      buttonRef.current.click();
+    }, 2000);
+
+    return () => {
+      clearInterval(interval);
+    };
+  }, []);
```

Comment:
@atapas , Shouldn't we guard this logic, because `buttonRef.current` might be null?
Also `buttonRef.current` is missing from the dependency list.

Reply:
The guard is not required as the buttonRef.current will always have values because the DOM structure will be prepared before our effect runs.

> Also buttonRef.current is missing from the dependency list.

This is interesting. Why should we pass ref as a dependency, as it is not a custom hook, and the ref is not changing? Whatever I read, we shouldn't do it for ref without a strong reason. Check this out: https://epicreact.dev/why-you-shouldnt-put-refs-in-a-dependency-array/

This is a good thing to decide on.

@Sachin-chaurasiya

Reply:
Interesting read 👍🏻, thanks for sharing it @atapas.
I got the point when we should include the ref in deps list.

Reply:
Thanks! Resolving.